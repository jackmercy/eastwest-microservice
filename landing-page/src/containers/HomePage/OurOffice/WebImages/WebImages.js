import { memo } from 'react';
import PropTypes from 'prop-types';

import useStyles from './webImages.styles';

const WebImages = ({ images }) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      {images.map((item) => (
        <div key={`our-office-image-${item.key}`} className={classes.item}>
          <img
            alt="Eastwest office"
            src={item.src}
            srcSet={item.srcSet}
            data-test="web-images-image"
            className={classes.image}
          />
        </div>
      ))}
    </div>
  );
};
WebImages.propTypes = {
  images: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string.isRequired,
      src: PropTypes.string.isRequired,
      srcSet: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default memo(WebImages);
