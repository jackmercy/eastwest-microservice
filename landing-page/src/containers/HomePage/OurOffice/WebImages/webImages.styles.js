import { createUseStyles } from 'react-jss';

import { rem } from '../../../../utils/jss';

export default createUseStyles({
  root: {
    margin: rem(-16),
    display: 'flex',
    flexWrap: 'wrap',
    transition: 'margin .5s ease',
  },
  item: {
    padding: rem(16),
    flex: '1 1 auto',
    '&:nth-of-child(3n + 1)': { flex: '2 1 auto' },
  },
  image: {
    display: 'block',
    margin: [0, 'auto'],
    maxHeight: rem(382),
    borderRadius: rem(15),
  },
});
