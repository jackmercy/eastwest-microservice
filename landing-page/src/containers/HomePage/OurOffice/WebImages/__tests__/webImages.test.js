import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import WebImages from '../WebImages';

describe('Testing HomePage > OurOffice > WebImages', () => {
  let wrapper;
  const props = {
    images: [
      {
        key: 'cover-1',
        src: '/static/images/homepage/office-01.png',
        srcSet: [
          '/static/images/homepage/office-01.png 500w',
          '/static/images/homepage/office-01@2x.png 1500w',
          '/static/images/homepage/office-01@3x.png 2000w',
        ].join(', '),
      },
      {
        key: 'cover-2',
        src: '/static/images/homepage/office-02.png',
        srcSet: [
          '/static/images/homepage/office-02.png 500w',
          '/static/images/homepage/office-02@2x.png 1500w',
          '/static/images/homepage/office-02@3x.png 2000w',
        ].join(', '),
      },
    ],
  };

  beforeAll(() => {
    wrapper = mount(<WebImages {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<WebImages {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have some images', () => {
    expect(wrapper.find('[data-test="web-images-image"]')).toHaveLength(
      props.images.length,
    );
  });
});
