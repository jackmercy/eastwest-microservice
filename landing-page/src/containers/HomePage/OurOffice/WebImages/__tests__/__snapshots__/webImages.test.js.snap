// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomePage > OurOffice > WebImages should render match snapshot 1`] = `
<div
  className="root-0-2-1"
>
  <div
    className="item-0-2-2"
    key="our-office-image-cover-1"
  >
    <img
      alt="Eastwest office"
      className="image-0-2-3"
      data-test="web-images-image"
      src="/static/images/homepage/office-01.png"
      srcSet="/static/images/homepage/office-01.png 500w, /static/images/homepage/office-01@2x.png 1500w, /static/images/homepage/office-01@3x.png 2000w"
    />
  </div>
  <div
    className="item-0-2-2"
    key="our-office-image-cover-2"
  >
    <img
      alt="Eastwest office"
      className="image-0-2-3"
      data-test="web-images-image"
      src="/static/images/homepage/office-02.png"
      srcSet="/static/images/homepage/office-02.png 500w, /static/images/homepage/office-02@2x.png 1500w, /static/images/homepage/office-02@3x.png 2000w"
    />
  </div>
</div>
`;
